File: maximum-number-of-words-found-in-sentences-2114/Solution.java
/**
 * LeetCode problem # 2114, 'Maximum Number of Words Found in Sentences'
 * https://leetcode.com/problems/maximum-number-of-words-found-in-sentences/
 */

class Solution {
    public int mostWordsFound(String[] sentences) {
        int res = 0;
        for (String sentence : sentences) {
            int count = 0;
            for (int i = 0; i < sentence.length(); ++i) {
                if (sentence.charAt(i) == ' ') {
                    ++count;
                }
            }
            res = Math.max(count + 1, res);
        }
        return res;

    }
}

File: maximum-number-of-words-found-in-sentences-2114/solution.ts
/**
 * LeetCode problem # 2114, 'Maximum Number of Words Found in Sentences'
 * https://leetcode.com/problems/maximum-number-of-words-found-in-sentences/
 */

function mostWordsFound(sentences: string[]): number {
    let res = 0;
    function countBlanks(s) {
        let res = 0;
        for (const char of s) {
            if (char === ' ') {
                ++res;
            }
        }
        return res;
    }

    for (const s of sentences) {
        res = Math.max(res, countBlanks(s) + 1);
    }

    return res;
};
